cmd/branchingconfigmanagers/bugzilla-config-manager/main.go

```diff
@@ -0,0 +1,236 @@
+}
+
+func extractFromConfig(lifecycleConfig ocplifecycle.Config, now time.Time) (developmentVersion *ocplifecycle.MajorMinor, gaVersions, nonEOLNonGAVersions []ocplifecycle.MajorMinor, err error) {
+	var errs []error
+	var developmentVersionFound bool
+
+	allNonEOLVersions := sets.String{}
+	gaVersionsSet := sets.String{}
+	for ocpProduct, productConfig := range lifecycleConfig {
+		if ocpProduct != "ocp" {
+			continue
+		}
+
+		for productVersion, events := range productConfig {
+			allNonEOLVersions.Insert(productVersion)
+			for _, event := range events {
+
+				// Future events, ignore
+				if event.When == nil || now.Before(event.When.Time) {
+					continue
+				}
+
+				// End of life version, nothing to do.
+				if event.Event == ocplifecycle.LifecycleEventEndOfLife {
+					allNonEOLVersions.Delete(productVersion)
+					break
+				}
+
+				if event.Event == ocplifecycle.LifecycleEventGenerallyAvailable {
+					gaVersionsSet.Insert(productVersion)
+					break
+				}
+
+				if event.Event == ocplifecycle.LifecycleEventOpen {
+					if developmentVersionFound {
+						errs = append(errs, fmt.Errorf("found multiple development versions: %s and %s", developmentVersion, productVersion))
+					}
+					parsedVersion, err := ocplifecycle.ParseMajorMinor(productVersion)
+					if err != nil {
+						errs = append(errs, fmt.Errorf("failed to parse %s as majorMinor version: %w", productVersion, err))
+						continue
+					}
+					developmentVersion = parsedVersion
+					developmentVersionFound = true
+				}
+
+			}
+		}
+	}
+
+	for _, nonEOLVersion := range allNonEOLVersions.List() {
+		parsed, err := ocplifecycle.ParseMajorMinor(nonEOLVersion)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to parse %s as majorMinor: %w", nonEOLVersion, err))
+			continue
+		}
+		if gaVersionsSet.Has(nonEOLVersion) {
+			gaVersions = append(gaVersions, *parsed)
+		} else {
+			nonEOLNonGAVersions = append(nonEOLNonGAVersions, *parsed)
+		}
+	}
+
+	return developmentVersion, gaVersions, nonEOLNonGAVersions, utilerrors.NewAggregate(errs)
+}
+
+func reconcileConfig(cfg *plugins.Bugzilla, developmentVersion *ocplifecycle.MajorMinor, gaVersions, nonEOLNonGAVersions []ocplifecycle.MajorMinor) *plugins.Bugzilla {
+	if cfg.Default == nil {
+		cfg.Default = map[string]plugins.BugzillaBranchOptions{}
+	}
+
+	if developmentVersion != nil {
+		// main/master: Set developmentVersion and leave the rest as is
+		for _, developmentBranchName := range []string{"main", "master"} {
+			config := cfg.Default[developmentBranchName]
+			config.TargetRelease = utilpointer.String(developmentVersion.String() + ".0")
+
+			cfg.Default[developmentBranchName] = config
+		}
+
+		// nonEOLNonGAVerion: Set config for {release/openshift}-$VERSION:
+		// * dependentBugStatus: ["MODIFIED", "ON_QA", "VERIFIED"]
+		// * DependentBugTargetReleases: "$VERSION+1.0"
+		// * ValidateByDefault: true
+		for _, nonEOLNonGAVerion := range nonEOLNonGAVersions {
```

Comment:
nit: why is this under `if developmentVersion != nil` ?

Reply:
Good catch, fixed